libs/wallet/src/web3-react/updater.ts

```diff
@@ -61,44 +66,83 @@ function _useWalletDetails(account?: string, standaloneMode?: boolean): WalletDe
   }, [isSmartContractWallet, walletName, icon, ensName])
 }
 
+
 function _useSafeInfo(walletInfo: WalletInfo): GnosisSafeInfo | undefined {
   const { provider } = useWeb3React()
   const { account, chainId } = walletInfo
   const [safeInfo, setSafeInfo] = useState<GnosisSafeInfo>()
-  const { isReadOnly } = useSafeAppsSdkInfo() || {}
+  const safeAppsSdk = useSafeAppsSdk()
 
   useEffect(() => {
-    const update = () => {
+    const updateSafeInfo = async () => {
       if (chainId && account && provider) {
-        getSafeInfo(chainId, account, provider)
-          .then((_safeInfo) =>
-            setSafeInfo({
+        if (safeAppsSdk) {
+          const appsSdkSafeInfo = await safeAppsSdk.safe.getInfo()
+          setSafeInfo((prevSafeInfo) => {
+            const { isReadOnly } = appsSdkSafeInfo
+            if (!prevSafeInfo) {
+              // We don't have prevSafeInfo for counterfactual safes as the SDK can't find a deployment
+              // of this safe on-chain
+              const { safeAddress, threshold, owners } = appsSdkSafeInfo
+              return {
+                chainId,
+                address: safeAddress,
+                threshold,
+                owners,
+                isReadOnly,
+              }
+            }
+
+            return {
+              ...prevSafeInfo,
               isReadOnly,
-              ..._safeInfo,
-            })
-          )
-          .catch(() => {
-            console.debug(`[WalletUpdater] Address ${account} is likely not a Safe (API didn't return Safe info)`)
-            setSafeInfo(undefined)
+            }
           })
+        } else {
+          getSafeInfo(chainId, account, provider)
+            .then((_safeInfo) => {
+              const { address, threshold, owners } = _safeInfo
+              setSafeInfo((prevSafeInfo) => ({
+                ...prevSafeInfo,
+                chainId,
+                address,
+                threshold,
+                owners,
+                isReadOnly: false,
+              }))
+
+              // Being here means that we don't run in an iframe over Safe's AppCommunicator,
+              // so we need to slow down the interval to not spam the RPC
+              clearInterval(shortSafeInfoInterval);
+              longSafeInfoInterval = setInterval(updateSafeInfo, SAFE_INFO_LONG_INTERVAL);
```

Comment:
I wrote somewhere (maybe another PR?) that it would be nice to only create the intervals conditionally, right? 

I mean, if `chainId, account, provider` changes, the whole hook will be destroyed and re-rendered creating a new interval. Why would we want to execute at regular intervals the check if `chainId && account && provider` is false?